### ISSUE TITLE
No matching distribution found for `ansible-core>=2.16.0` when running `ansible-lint` as pre-commit hook

### ISSUE THREAD

**Issue by CM000n** [ansible/ansible-lint#4616]
##### Summary

When I try to run `ansible-lint` as `pre-commit` hook, the installation of the hook envronment fails every time with the error message `No matching distribution found for ansible-core>=2.16.0`

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

Windows Subsystem for Linux (WSL2)
```console
➜  lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 24.04.2 LTS
Release:        24.04
Codename:       noble
```

This is my corresponding `.pre-commit-config.yaml` file:
```yaml
default_language_version:
    python: python3.11
repos:
  - repo: https://github.com/pre-commit/mirrors-isort
    rev: v5.10.1
    hooks:
      - id: isort
  - repo: https://github.com/psf/black
    rev: 25.1.0
    hooks:
      - id: black
  - repo: https://github.com/python-poetry/poetry
    rev: 2.1.3
    hooks:
      - id: poetry-check
        name: 📜 Check pyproject with Poetry
  - repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v5.0.0
    hooks:
      - id: check-json
        name: ✅ Check JSON files
        exclude: .devcontainer
      - id: check-toml
        name: ✅ Check TOML files
      - id: check-yaml
        name: ✅ Check YAML files
      - id: check-added-large-files
        name: 📁 Check for large files
      - id: debug-statements
        name: 🐛 Check for forgotten debug statemens
      - id: check-symlinks
        name: 🔗 Check for broken symlinks
      - id: check-merge-conflict
        name: 💥 Check for merge conflicts
      - id: end-of-file-fixer
        name: 🔚 Fix End of Files
      - id: trailing-whitespace
        name: ⬜ Fix trailing whitespaces
        exclude_types: [text]
      - id: requirements-txt-fixer
        name: 📝 Fix requirements.txt
  - repo: https://github.com/IamTheFij/ansible-pre-commit
    rev: v1.0.1
    hooks:
      - id: encryption-check
        name: 🔐 Encryption check
        exclude: "vault-client.sh"
  - repo: https://github.com/ansible-community/ansible-lint.git
    rev: v25.5.0
    hooks:
      - id: ansible-lint
        name: 🅰️ Ansible Linter
        files: \.(yaml|yml)$
        entry: bash -c 'exec env ANSIBLE_CONFIG=ansible-lint/ansible.cfg bash -c "env | python3 -m ansiblelint -v --force-color --exclude ~/.ansible/ --exclude .gitlab-ci.yml --exclude .pre-commit-config.yaml"'

```

- ansible installation method: poetry
- ansible-lint installation method: pre-commit

##### STEPS TO REPRODUCE

1. Set up pre-commit with ansible-lint pre-commit hook in `.pre-commit-config.yaml`
2. Let pre-commit initialize and installing hook environments by running `pre-commit run --all-files --verbose`

##### Desired Behavior

pre-commit should initialize and install hook environment without problems

##### Actual Behavior

The attempt to install a `pre-commit` hook environment for `ansible-lint` results in the following error:
```console
➜  pre-commit clean                    
Cleaned /home/d0286806/.cache/pre-commit.

➜  pre-commit run --all-files --verbose
[INFO] Initializing environment for https://github.com/pre-commit/mirrors-isort.
[INFO] Initializing environment for https://github.com/psf/black.
[INFO] Initializing environment for https://github.com/python-poetry/poetry.
[INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Initializing environment for https://github.com/IamTheFij/ansible-pre-commit.
[INFO] Initializing environment for https://github.com/ansible-community/ansible-lint.git.
[INFO] Initializing environment for https://github.com/ansible-community/ansible-lint.git:ansible-core>=2.16.0.
[INFO] Installing environment for https://github.com/pre-commit/mirrors-isort.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/psf/black.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/python-poetry/poetry.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Installing environment for https://github.com/ansible-community/ansible-lint.git.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('/home/d0286806/.cache/pre-commit/repog4memw_0/py_env-python3.11/bin/python', '-mpip', 'install', '.', 'ansible-core>=2.16.0')
return code: 1
stdout:
    Processing /home/d0286806/.cache/pre-commit/repog4memw_0
      Installing build dependencies: started
      Installing build dependencies: finished with status 'done'
      Getting requirements to build wheel: started
      Getting requirements to build wheel: finished with status 'done'
      Preparing metadata (pyproject.toml): started
      Preparing metadata (pyproject.toml): finished with status 'done'
stderr:
    ERROR: Ignored the following yanked versions: 2.14.0rc1.post0
    ERROR: Ignored the following versions that require a different python version: 2.16.0 Requires-Python >=3.10; 2.16.0b1 Requires-Python >=3.10; 2.16.0b2 Requires-Python >=3.10; 2.16.0rc1 Requires-Python >=3.10; 2.16.1 Requires-Python >=3.10; 2.16.10 Requires-Python >=3.10; 2.16.10rc1 Requires-Python >=3.10; 2.16.11 Requires-Python >=3.10; 2.16.11rc1 Requires-Python >=3.10; 2.16.12 Requires-Python >=3.10; 2.16.12rc1 Requires-Python >=3.10; 2.16.13 Requires-Python >=3.10; 2.16.13rc1 Requires-Python >=3.10; 2.16.14 Requires-Python >=3.10; 2.16.14rc1 Requires-Python >=3.10; 2.16.1rc1 Requires-Python >=3.10; 2.16.2 Requires-Python >=3.10; 2.16.3 Requires-Python >=3.10; 2.16.3rc1 Requires-Python >=3.10; 2.16.4 Requires-Python >=3.10; 2.16.4rc1 Requires-Python >=3.10; 2.16.5 Requires-Python >=3.10; 2.16.5rc1 Requires-Python >=3.10; 2.16.6 Requires-Python >=3.10; 2.16.7 Requires-Python >=3.10; 2.16.7rc1 Requires-Python >=3.10; 2.16.8 Requires-Python >=3.10; 2.16.8rc1 Requires-Python >=3.10; 2.16.9 Requires-Python >=3.10; 2.16.9rc1 Requires-Python >=3.10; 2.17.0 Requires-Python >=3.10; 2.17.0b1 Requires-Python >=3.10; 2.17.0rc1 Requires-Python >=3.10; 2.17.0rc2 Requires-Python >=3.10; 2.17.1 Requires-Python >=3.10; 2.17.10 Requires-Python >=3.10; 2.17.10rc1 Requires-Python >=3.10; 2.17.11 Requires-Python >=3.10; 2.17.11rc1 Requires-Python >=3.10; 2.17.12 Requires-Python >=3.10; 2.17.12rc1 Requires-Python >=3.10; 2.17.1rc1 Requires-Python >=3.10; 2.17.2 Requires-Python >=3.10; 2.17.2rc1 Requires-Python >=3.10; 2.17.2rc2 Requires-Python >=3.10; 2.17.3 Requires-Python >=3.10; 2.17.3rc1 Requires-Python >=3.10; 2.17.4 Requires-Python >=3.10; 2.17.4rc1 Requires-Python >=3.10; 2.17.5 Requires-Python >=3.10; 2.17.5rc1 Requires-Python >=3.10; 2.17.6 Requires-Python >=3.10; 2.17.6rc1 Requires-Python >=3.10; 2.17.7 Requires-Python >=3.10; 2.17.7rc1 Requires-Python >=3.10; 2.17.8 Requires-Python >=3.10; 2.17.8rc1 Requires-Python >=3.10; 2.17.9 Requires-Python >=3.10; 2.17.9rc1 Requires-Python >=3.10; 2.18.0 Requires-Python >=3.11; 2.18.0b1 Requires-Python >=3.11; 2.18.0rc1 Requires-Python >=3.11; 2.18.0rc2 Requires-Python >=3.11; 2.18.1 Requires-Python >=3.11; 2.18.1rc1 Requires-Python >=3.11; 2.18.2 Requires-Python >=3.11; 2.18.2rc1 Requires-Python >=3.11; 2.18.3 Requires-Python >=3.11; 2.18.3rc1 Requires-Python >=3.11; 2.18.4 Requires-Python >=3.11; 2.18.4rc1 Requires-Python >=3.11; 2.18.5 Requires-Python >=3.11; 2.18.5rc1 Requires-Python >=3.11; 2.18.6 Requires-Python >=3.11; 2.18.6rc1 Requires-Python >=3.11; 2.19.0b1 Requires-Python >=3.11; 2.19.0b2 Requires-Python >=3.11; 2.19.0b3 Requires-Python >=3.11; 2.19.0b4 Requires-Python >=3.11
    ERROR: Could not find a version that satisfies the requirement ansible-core>=2.16.0 (from versions: 0.0.1a1, 2.11.0b1, 2.11.0b2, 2.11.0b3, 2.11.0b4, 2.11.0rc1, 2.11.0rc2, 2.11.0, 2.11.1rc1, 2.11.1, 2.11.2rc1, 2.11.2, 2.11.3rc1, 2.11.3, 2.11.4rc1, 2.11.4, 2.11.5rc1, 2.11.5, 2.11.6rc1, 2.11.6, 2.11.7rc1, 2.11.7, 2.11.8rc1, 2.11.8, 2.11.9rc1, 2.11.9, 2.11.10rc1, 2.11.10, 2.11.11rc1, 2.11.11, 2.11.12rc1, 2.11.12, 2.12.0b1, 2.12.0b2, 2.12.0rc1, 2.12.0, 2.12.1rc1, 2.12.1, 2.12.2rc1, 2.12.2, 2.12.3rc1, 2.12.3, 2.12.4rc1, 2.12.4, 2.12.5rc1, 2.12.5, 2.12.6rc1, 2.12.6, 2.12.7rc1, 2.12.7, 2.12.8rc1, 2.12.8, 2.12.9rc1, 2.12.9, 2.12.10rc1, 2.12.10, 2.13.0b0, 2.13.0b1, 2.13.0rc1, 2.13.0, 2.13.1rc1, 2.13.1, 2.13.2rc1, 2.13.2, 2.13.3rc1, 2.13.3, 2.13.4rc1, 2.13.4, 2.13.5rc1, 2.13.5, 2.13.6rc1, 2.13.6, 2.13.7rc1, 2.13.7, 2.13.8rc1, 2.13.8, 2.13.9rc1, 2.13.9, 2.13.10rc1, 2.13.10, 2.13.11rc1, 2.13.11, 2.13.12rc1, 2.13.12, 2.13.13rc1, 2.13.13, 2.14.0b1, 2.14.0b2, 2.14.0b3, 2.14.0rc1, 2.14.0rc2, 2.14.0, 2.14.1rc1, 2.14.1, 2.14.2rc1, 2.14.2, 2.14.3rc1, 2.14.3, 2.14.4rc1, 2.14.4, 2.14.5rc1, 2.14.5, 2.14.6rc1, 2.14.6, 2.14.7rc1, 2.14.7, 2.14.8rc1, 2.14.8, 2.14.9rc1, 2.14.9, 2.14.10rc1, 2.14.10, 2.14.11rc1, 2.14.11, 2.14.12rc1, 2.14.12, 2.14.13, 2.14.14rc1, 2.14.14, 2.14.15rc1, 2.14.15, 2.14.16rc1, 2.14.16, 2.14.17rc1, 2.14.17, 2.14.18rc1, 2.14.18, 2.15.0b1, 2.15.0b2, 2.15.0b3, 2.15.0rc1, 2.15.0rc2, 2.15.0, 2.15.1rc1, 2.15.1, 2.15.2rc1, 2.15.2, 2.15.3rc1, 2.15.3, 2.15.4rc1, 2.15.4, 2.15.5rc1, 2.15.5, 2.15.6rc1, 2.15.6, 2.15.7rc1, 2.15.7, 2.15.8, 2.15.9rc1, 2.15.9, 2.15.10rc1, 2.15.10, 2.15.11rc1, 2.15.11, 2.15.12rc1, 2.15.12, 2.15.13rc1, 2.15.13)
    ERROR: No matching distribution found for ansible-core>=2.16.0
```

**Comment by CM000n** [comment]
Hi @alisonlhart, thanks for asking. `pre-commit` manages the environment of its hooks itself.
But as you can see from the pre-commit config, the `default_language_version` for pre-commit is `python3.11`.
You can also see in the error message that the environemnt used is located under '/home/d0286806/.cache/pre-commit/repog4memw_0/py_env-python3.11/bin/python' and is therefore a Python 3.11 environment.

**Comment by CM000n** [comment]
Edit
Ok it looks like it was an isue with the venv after all. 
My default python version on the system was 3.9. After I changed it to 3.11 and ran `pre-commit clean` and re-initialized the hooks, it worked.
Sory for the noise, and thanks again
